core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -167,20 +170,29 @@ private class AsyncEventQueue(
     }
     logTrace(s"Dropping event $event")
 
-    val droppedCount = droppedEventsCounter.get
+    val droppedCount = droppedEventsCounter.get - lastDroppedEventsCounter
+    val lastReportTime = lastReportTimestamp.get
+    val curTime = System.currentTimeMillis()
     if (droppedCount > 0) {
       // Don't log too frequently
-      if (System.currentTimeMillis() - lastReportTimestamp >= 60 * 1000) {
-        // There may be multiple threads trying to decrease droppedEventsCounter.
-        // Use "compareAndSet" to make sure only one thread can win.
-        // And if another thread is increasing droppedEventsCounter, "compareAndSet" will fail and
-        // then that thread will update it.
-        if (droppedEventsCounter.compareAndSet(droppedCount, 0)) {
-          val prevLastReportTimestamp = lastReportTimestamp
-          lastReportTimestamp = System.currentTimeMillis()
+      if (curTime - lastReportTime >= LOGGING_INTERVAL) {
+        // There may be multiple threads trying to logging dropped events,
+        // Use 'compareAndSet' to make sure only one thread can win.
+        // After set the 'lastReportTimestamp', the next time we come here will
+        // be 60s later.
+        if (lastReportTimestamp.compareAndSet(lastReportTime, curTime)) {
+          val prevLastReportTimestamp = lastReportTimestamp.get
           val previous = new java.util.Date(prevLastReportTimestamp)
+          lastDroppedEventsCounter = droppedCount
           logWarning(s"Dropped $droppedCount events from $name since " +
             s"${if (prevLastReportTimestamp == 0) "the application started" else s"$previous"}.")
+          // Logging thread dump when events from appStatus was dropped
```

Comment:
I'm not sure how much this helps; as current event being processed may not be culprit to lag the overall executions. 

Spark will introduce a new configuration to log the event and listener name if the execution takes more than configured threshold; it would be more helpful. The commit is 
https://github.com/apache/spark/commit/0346afa8fc348aa1b3f5110df747a64e3b2da388

Reply:
Agree that this log is less useful than it might seem at first.